--- a/os-probes/mounted/common/90linux-distro	2017-05-04 14:57:23.197520770 +0800
+++ b/os-probes/mounted/common/90linux-distro	2017-05-04 15:09:13.848541790 +0800
@@ -137,6 +137,9 @@ if (ls "$dir"/lib*/ld*.so* && [ -d "$dir
 	elif [ -e "$dir/etc/devuan_version" ]; then
 		short="Devuan"
 		long="$(printf "Devuan GNU/Linux (%s)\n" "$(cat "$dir/etc/devuan_version")")"
+	elif [ -e "$dir/usr/bin/crux" ]; then
+		short="CRUX"
+		long="$(printf "CRUX GNU/Linux" "$(cat "$dir/usr/bin/crux" | grep echo | cut -b 20-22)")"
 	else
 		short="Linux"
 		long="unknown Linux distribution"
