Index: Vulkan-ValidationLayers-sdk-1.1.121.0/CMakeLists.txt
===================================================================
--- Vulkan-ValidationLayers-sdk-1.1.121.0/CMakeLists.txt
+++ Vulkan-ValidationLayers-sdk-1.1.121.0/CMakeLists.txt	2019-09-25 12:25:23.778410890 +0200
@@ -27,6 +27,9 @@
 
 enable_testing()
 
+set(VKLAYERS_VERSION "1.0.0" CACHE STRING "Build version")
+set(VKLAYERS_SOVERSION "1" CACHE STRING "Library API version")
+
 # User-interface declarations ----------------------------------------------------------------------------------------------------
 # This section contains variables that affect development GUIs (e.g. CMake GUI and IDEs), such as option(), folders, and variables
 # with the CACHE property.
Index: Vulkan-ValidationLayers-sdk-1.1.121.0/layers/CMakeLists.txt
===================================================================
--- Vulkan-ValidationLayers-sdk-1.1.121.0/layers/CMakeLists.txt
+++ Vulkan-ValidationLayers-sdk-1.1.121.0/layers/CMakeLists.txt	2019-09-25 12:25:23.779410888 +0200
@@ -133,6 +133,9 @@
 else(UNIX AND NOT APPLE) # i.e.: Linux
     macro(AddVkLayer target LAYER_COMPILE_DEFINITIONS)
         add_library(VkLayer_${target} SHARED ${ARGN})
+        set_target_properties(VkLayer_${target} PROPERTIES
+                              VERSION ${VKLAYERS_VERSION}
+                              SOVERSION ${VKLAYERS_SOVERSION})
         target_compile_definitions(VkLayer_${target} PUBLIC ${LAYER_COMPILE_DEFINITIONS})
         target_link_libraries(VkLayer_${target} PRIVATE VkLayer_utils)
         add_dependencies(VkLayer_${target} VkLayer_utils)
