diff -Naur clucene-core-2.3.3.4-old/CMakeLists.txt clucene-core-2.3.3.4/CMakeLists.txt
--- clucene-core-2.3.3.4-old/CMakeLists.txt	2018-12-30 01:11:38.564342744 +0100
+++ clucene-core-2.3.3.4/CMakeLists.txt	2018-12-30 01:11:48.033493007 +0100
@@ -13,7 +13,7 @@
 #Must be incremented for releases if the api is not backwards compatible
 SET(CLUCENE_SOVERSION "1")
 
-MATH(EXPR CLUCENE_INT_VERSION "(${CLUCENE_VERSION_MAJOR} * 1000000) + (${CLUCENE_VERSION_MINOR} * 10000) + (${CLUCENE_VERSION_REVISION} * 100) + (${CLUCENE_VERSION_PATCH} * 1)" )
+#MATH(EXPR CLUCENE_INT_VERSION "(${CLUCENE_VERSION_MAJOR} * 1000000) + (${CLUCENE_VERSION_MINOR} * 10000) + (${CLUCENE_VERSION_REVISION} * 100) + (${CLUCENE_VERSION_PATCH} * 1)" )
 SET(CLUCENE_VERSION "${CLUCENE_VERSION_MAJOR}.${CLUCENE_VERSION_MINOR}.${CLUCENE_VERSION_REVISION}.${CLUCENE_VERSION_PATCH}")
 
 #CMake 2.6+ is recommended to an improved Boost module
@@ -60,11 +60,11 @@
 OPTION(ENABLE_ASCII_MODE
   "enable ascii support"
   OFF)
-  
+
 SET(ENABLE_ANSI_MODE OFF)
 IF(CMAKE_COMPILER_IS_GNUCXX)
   SET(ENABLE_ANSI_MODE ON)
-  
+
   #exceptions:
   IF(MINGW OR CYGWIN)
     SET(ENABLE_ANSI_MODE OFF)
@@ -123,7 +123,7 @@
             SET(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -pg")
         ENDIF ( ENABLE_GPROF )
     ENDIF ( GccFlagPg )
-    
+
     IF("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
         SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC"  )
     ENDIF("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
@@ -131,7 +131,7 @@
    IF( ENABLE_ANSI_MODE )
     SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ansi")
    ENDIF ( ENABLE_ANSI_MODE )
-ENDIF(CMAKE_COMPILER_IS_GNUCC) 
+ENDIF(CMAKE_COMPILER_IS_GNUCC)
 
 
 #Single output directory for building all executables and libraries.
@@ -172,7 +172,7 @@
   "${CMAKE_MODULE_PATH}/cmake_uninstall.cmake.in"
   "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
   IMMEDIATE @ONLY)
-  
+
 ADD_CUSTOM_TARGET(uninstall
   "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
 
diff -Naur clucene-core-2.3.3.4-old/src/shared/cmake/MacroEnsureVersion.cmake clucene-core-2.3.3.4/src/shared/cmake/MacroEnsureVersion.cmake
--- clucene-core-2.3.3.4-old/src/shared/cmake/MacroEnsureVersion.cmake	2018-12-30 01:11:38.579342982 +0100
+++ clucene-core-2.3.3.4/src/shared/cmake/MacroEnsureVersion.cmake	2018-12-30 01:12:53.301528727 +0100
@@ -24,7 +24,7 @@
 
     # compute an overall version number which can be compared at once
     MATH(EXPR req_vers_num "${req_major_vers}*10000 + ${req_minor_vers}*100 + ${req_patch_vers}")
-    MATH(EXPR found_vers_num "${found_major_vers}*10000 + ${found_minor_vers}*100 + ${found_patch_vers}")
+    #MATH(EXPR found_vers_num "${found_major_vers}*10000 + ${found_minor_vers}*100 + ${found_patch_vers}")
 
     if (found_vers_num LESS req_vers_num)
         set( ${var_too_old} FALSE )
