diff --git a/meson.build b/meson.build
index 6e35c00f77aef272a20c3d473ce50bed0206a08c..28799d8bec46aef4d1b05b44ef557cd387f48a64 100644
--- a/meson.build
+++ b/meson.build
@@ -43,7 +43,7 @@ pkg_deps = [
   {'name': 'glib-2.0', 'version': '>= 2.32' },
   {'name': 'gobject-2.0', 'version': '>= 2.13.0' },
   {'name': 'gtk+-3.0', 'version': '>= 3.22.0' },
-  {'name': STARTUP_NOTIFICATION_PACKAGE, 'version': '>= 0.4', 'required': get_option('startup_notification').enabled() },
+  {'name': STARTUP_NOTIFICATION_PACKAGE, 'version': '>= 0.4', 'required': get_option('startup_notification')},
   {'name': X11_PACKAGE },
   {'name': XRES_PACKAGE, 'required': false},
 ]
@@ -54,7 +54,7 @@ foreach p: pkg_deps
   dep = dependency(pkg, version: p.get('version', []), required: required)
   LIBWNCK_DEPS += dep
 
-  if not required and dep.found()
+  if dep.found()
     conf.set('HAVE_' + pkg.to_upper().underscorify(), 1)
   endif
 endforeach
