--- asterisk-13.13.1/menuselect-tree.orig	2017-01-30 11:04:27.059563497 +0900
+++ asterisk-13.13.1/menuselect-tree	2017-01-30 11:04:39.656181039 +0900
@@ -1920,7 +1920,7 @@
 </category>
 <category name="MENUSELECT_UTILS" displayname="Utilities" positive_output="yes">
   <member name="aelparse">
-	<defaultenabled>no</defaultenabled>
+	<defaultenabled>yes</defaultenabled>
 	<support_level>extended</support_level>
   </member>
   <member name="astcanary">
